### PR TITLE
Add parameter placeholders when autocompleting method

### PR DESCRIPTION
**Pull request by iluuu1994**
Screen Shot 2019-06-23 at 14 58 10

Questions:

- [ ] Does this need a fallback? [The LSP spec](https://microsoft.github.io/language-server-protocol/specification) defines a `snippetSupport` flag. I have not checked if most LSP implementations implement snippets.
- [ ] Should we add an option to disable this behaviour? Not everybody might likes this.

### CODE REVIEW

File: src/Psalm/Codebase.php [hunk 1]
```diff
@@ -1229,8 +1230,28 @@ public function getCompletionItemsForClassishThing(string $type_string, string $
                             null,
                             (string)$method_storage->visibility,
                             $method_storage->cased_name,
-                            $method_storage->cased_name . '()'
+                            $method_storage->cased_name . '(' . implode(
+                                ', ',
+                                array_map(function (FunctionLikeParameter $parameter) {
+                                    /** @var int $i */
+                                    static $i = 0;
```
Comment (psalm-shepherd) [inline src/Psalm/Codebase.php]: Left operand cannot be mixed
```
                                    ++$i;
                                      ^^
```

File: src/Psalm/Codebase.php [hunk 2]
```diff
@@ -1229,8 +1230,28 @@ public function getCompletionItemsForClassishThing(string $type_string, string $
...
+                                    ++$i;
+                                    return '${'
```
Comment (psalm-shepherd) [inline src/Psalm/Codebase.php]: Right operand cannot be mixed
```
                                        . $i
                                          ^^
```

File: src/Psalm/Codebase.php [hunk 3]
```diff
@@ -1229,8 +1230,28 @@ public function getCompletionItemsForClassishThing(string $type_string, string $
...
+                                        . $i
+                                        . ':'
+                                        . (
+                                            $parameter->type !== null
```
Comment (psalm-shepherd) [inline src/Psalm/Codebase.php]: Argument 3 of str_replace expects array<array-key, mixed>|string, Psalm\Type\Union provided with a __toString method
```
                                                ? str_replace(['$', '}'], ['\$', '\}'], $parameter->type) . ' '
                                                                                        ^^^^^^^^^^^^^^^^
```

**Comment by psalm-shepherd** [comment]
Psalm also found errors in other files

```
src/Psalm/Codebase.php:1211:99 - Param #3 is never referenced in this method
```

**Comment by psalm-shepherd** [comment]
Psalm didn’t find any errors!

**Comment by iluuu1994** [comment]
Maybe it would be better to implement `SignatureHelp`. The only thing this PR should do then is to jump into the parentheses of the method.

**Comment by muglug** [comment]
Yeah I prefer `SignatureHelp` - feels like it’s the LSP-preferred way.

**Comment by iluuu1994** [comment]
Cool, I'll look into that.

**Pull request closed without merging**